src/components/players/background-player.tsx

```diff
@@ -0,0 +1,154 @@
+
+export type BackgroundPlayerProps = Omit<DefaultPlayerProps, 'src'>;
+
+export const BackgroundPlayer = forwardRef((allProps: BackgroundPlayerProps, forwardedRef: any) => {
+  let {
+    style,
+    children,
+    asset,
+    controls,
+    poster,
+    blurDataURL,
+    onPlaying,
+    onLoadStart,
+    thumbnailTime,
+    ...rest
+  } = allProps;
+
+  const slottedPoster = Children.toArray(children).find((child) => {
+    return typeof child === 'object' && 'type' in child && child.props.slot === 'poster';
+  });
+
+  // If there's a slotted poster image (e.g. next/image) remove the default player poster and blurDataURL.
+  if (isValidElement(slottedPoster)) {
+    poster = '';
+    blurDataURL = undefined;
+  }
+
+  const props = rest as MuxVideoProps;
+  const imgStyleProps: React.CSSProperties = {};
+  const playbackId = asset ? getPlaybackId(asset) : undefined;
+
+  let isCustomPoster = true;
+  let srcSet: string | undefined;
+
+  if (playbackId && asset?.status === 'ready') {
+    props.src = undefined;
+    props.playbackId = playbackId;
+
+    if (poster) {
+      isCustomPoster = poster !== getPosterURLFromPlaybackId(playbackId, props);
+
+      if (!isCustomPoster) {
+        // If it's not a custom poster URL, optimize with a srcset.
+        srcSet =
+          `${getPosterURLFromPlaybackId(playbackId, { ...props, thumbnailTime, width: 480 })} 480w,` +
+          `${getPosterURLFromPlaybackId(playbackId, { ...props, thumbnailTime, width: 640 })} 640w,` +
+          `${getPosterURLFromPlaybackId(playbackId, { ...props, thumbnailTime, width: 960 })} 960w,` +
+          `${getPosterURLFromPlaybackId(playbackId, { ...props, thumbnailTime, width: 1280 })} 1280w,` +
+          `${getPosterURLFromPlaybackId(playbackId, { ...props, thumbnailTime, width: 1600 })} 1600w,` +
+          `${getPosterURLFromPlaybackId(playbackId, { ...props, thumbnailTime })} 1920w`;
+      }
+    }
+  }
+
+  if (blurDataURL) {
+    const showGeneratedBlur = !isCustomPoster && blurDataURL === asset?.blurDataURL;
+    const showCustomBlur = isCustomPoster && blurDataURL !== asset?.blurDataURL;
+
+    if (showGeneratedBlur || showCustomBlur) {
+      imgStyleProps.color = 'transparent';
+      imgStyleProps.backgroundSize = 'cover';
+      imgStyleProps.backgroundPosition = 'center';
+      imgStyleProps.backgroundRepeat = 'no-repeat';
+      imgStyleProps.backgroundImage = `url('data:image/svg+xml;charset=utf-8,${svgBlurImage(blurDataURL)}')`;
+    }
+  }
+
+  let videoPoster;
+  if (poster) {
+    videoPoster = '';
+  }
+
+  const [posterHidden, setPosterHidden] = useState(false);
+
+  return (
+    <>
+      <style>{
+        /* css */`
+        .next-video-bg {
+          aspect-ratio: initial;
+          height: 100%;
+        }
+
+        .next-video-bg-poster,
+        .next-video-bg-text {
+          position: absolute;
+          top: 0;
+          left: 0;
+          width: 100%;
+          height: 100%;
+        }
+
+        .next-video-bg-poster {
```

Comment:
Is there a risk this behavior isn't the same as the underlying video? We probably always want this poster to seamlessly move into the first frame of the video.

Reply:
On that note, I don't see anything in here around defaulting to the first frame for the poster image.

Reply:
could you explain the first question a bit more?

the default Mux poster is set to thumbnailTime=0 here
https://github.com/muxinc/next-video/pull/207/files#diff-602b8d1f33f5c6a7125fa72ff12de7212fb09bcbdb94475b4b13a4fca5e5a674R16

the only thing that is not lining up is the blur up image because we don't have a good way to set a client prop to the server yet, I'm currently digging in Next.js source to pull this off